src/tools/cephfs/cephfs-shell

```diff
@@ -773,6 +773,148 @@ class CephFSShell(Cmd):
 
         return 0
 
+    def complete_ln(self, text, line, begidx, endidx):
+        """
+        auto complete of file name.
+        """
+        return self.complete_filenames(text, line, begidx, endidx)
+
+    ln_parser = argparse.ArgumentParser(
+        description='Create a hard link or a symbolic '
+                    'link to an existing file or directory.')
+    ln_parser.add_argument('src_path', type=str, action=path_to_bytes,
+                           help='Path of the source file')
+    ln_parser.add_argument('dest_path', type=str, action=path_to_bytes,
+                           help='Destination path', nargs='?')
+    ln_parser.add_argument('-s', '--symbolic', action='store_true',
+                           help='Create symbolic link')
+    ln_parser.add_argument('-v', '--verbose', action='store_true',
+                           help='Print name of each linked file')
+    ln_parser.add_argument('-f', '--force', action='store_true',
+                           help='Force create link/symbolic link')
+
+    @with_argparser(ln_parser)
+    def do_ln(self, args):
+        # check if source file/dir exists.
+        if not is_file_exists(args.src_path) \
+                and not is_dir_exists(args.src_path):
+            set_exit_code_msg(errno.ENOENT,
+                              msg=f"ln: failed to access "
+                                  f"'{args.src_path.decode('utf-8')}"
+                                  f"': No such file or directory")
+            return
+        src_path = args.src_path
```

Comment:
Better to use just `src` and `dest` than `src_path` and `dest_path` because it is obvious that both have to be paths?

Reply:
I forgot to post this yesterday, I think it will be more appropriate to use `target` and `link` instead.